### DISCUSSION TITLE
PNL Calculation

### DISCUSSION THREAD

**Discussion by pakornavakul** [nkaz001/hftbacktest#85] [Q&A] [answered]
Hello, 
Questions regarding how PNL is calculated. I seems like you are using mid, balance, position, and fee to form an equity curve. How does the actual filled price gets factor into this PNL? Through balance?

Thanks for your help!

**Comment by nkaz001** [comment, marked as answer]
yes, the balance is the cumulative value calculated by multiplying the filled price by the filled quantity, considering the side of execution.

please see https://github.com/nkaz001/hftbacktest/blob/8f1784b75caa651e6ce98d5e8ef6bf57ade1fc7d/hftbacktest/state.py#L25
